### PR TITLE
Annotate Head Node with {"head": 1.0}

### PR DESCRIPTION
**Pull request by simon-mo**
## Why are these changes needed?
This PR adds a static resource for head node, annotating with {"head":1}. This simplify many cases when we need to pin certain task or actor to the head node.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Review by richardliaw — COMMENTED** [review]
Please document this somewhere. also nit: `node_id_prefix:head`

**Review by edoakes — CHANGES_REQUESTED** [review]
This qualifies as an API change IMO, we should go through the normal process for it (get feedback and approval from other maintainers).

**Comment by richardliaw** [comment]
@simon-mo A short doc would be good (1 or 2 pager). FYI this may be frustrating for you because it is a minor change, but a doc + short review would allow us to identify a possible better alternative (or see if there's a larger fundamental issue).

**Comment by ericl** [comment]
I'm fine with this as long as it has node id prefix.

**Comment by simon-mo** [comment]
From offline discussion with @robertnishihara, we are going to close this PR for the following reasons:

- For long term direction, we don't want the concept of "head node" in Ray. Especially with the integration of GCS service, all nodes have the same status.
- For Serve's use case, what we need is a node with IP address user knows how to reach. Since user already knows which node do they want to place their server on, they can directly tell serve to place HTTP server on that specific node.
- For other use case like placing some actors in non-preemptible nodes, we can improve the autoscaler automatically tag the machines with machine types and preemptibility.

**Pull request closed without merging**
